### PR TITLE
Removes BSQL

### PR DESCRIPTION
**Pull request by Bibby0110**
### Intent of your Pull Request
Port from TG
I'll write more tomorrow, it's late

#### Changelog

:cl:  
rscadd: SQL queries should be more stable
/:cl:

### CODE REVIEW

File: code/__DEFINES/rust_g.dm [hunk 1]
```diff
@@ -21,6 +61,16 @@
 #define RUSTG_HTTP_METHOD_DELETE "delete"
 #define RUSTG_HTTP_METHOD_PATCH "patch"
 #define RUSTG_HTTP_METHOD_HEAD "head"
+#define rustg_http_request_blocking(method, url, body, headers) call(RUST_G, "http_request_blocking")(method, url, body, headers)
+#define rustg_http_request_async(method, url, body, headers) call(RUST_G, "http_request_async")(method, url, body, headers)
+#define rustg_http_check_request(req_id) call(RUST_G, "http_check_request")(req_id)
```
Comment (alexkar598) [inline code/__DEFINES/rust_g.dm]:
```suggestion
```

Already there

**Review by alexkar598 — DISMISSED** [review]

**Review by JamieD1 — APPROVED** [review]

File: .github/workflows/turdis.yml [hunk 2]
```diff
@@ -30,6 +30,13 @@ jobs:
       
       - name: Setup Rust
         uses: hecrj/setup-rust-action@v1.3.2
+        with:
+          default: true
+      
+      - name: Setup PHP
+        uses: nanasess/setup-php@v3.0.3
+        with: 
+          php_version: 7.2
```
Comment (alexkar598) [inline .github/workflows/turdis.yml]: Why did you bring back php?

File: code/controllers/subsystem/blackbox.dm [hunk 3]
```diff
@@ -45,7 +45,17 @@ SUBSYSTEM_DEF(blackbox)
...
-	var/datum/DBQuery/query_record_playercount = SSdbcore.NewQuery("INSERT INTO [format_table_name("legacy_population")] (playercount, admincount, time, server_ip, server_port, round_id) VALUES ([playercount], [admincount], '[SQLtime()]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', '[GLOB.round_id]')")
+	var/datum/DBQuery/query_record_playercount = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("legacy_population")] (playercount, admincount, time, server_ip, server_port, round_id)
+		VALUES (:playercount, :admincount, :time, INET_ATON(:server_ip), :server_port, :round_id)
+	"}, list(
+		"playercount" = playercount,
+		"admincount" = admincount,
+		"time" = SQLtime(),
+		"server_ip" = world.internet_address || "0",
+		"server_port" = "[world.port]",
```
Comment (alexkar598) [inline code/controllers/subsystem/blackbox.dm]: This is a SMALLINT not a string
```suggestion
		"server_port" = world.port,
```

File: code/controllers/subsystem/blackbox.dm [hunk 4]
```diff
@@ -286,51 +301,36 @@ Versioning
...
+		"clone" = L.getCloneLoss(),
+		"stamina" = L.getStaminaLoss(),
+		"x_coord" = L.x,
+		"y_coord" = L.y,
+		"z_coord" = L.z,
+		"last_words" = L.last_words,
+		"suicide" = L.suiciding,
+		"map" = SSmapping.config.map_name,
+		"internet_address" = world.internet_address || "0",
+		"port" = "[world.port]",
```
Comment (alexkar598) [inline code/controllers/subsystem/blackbox.dm]:
```suggestion
		"port" = world.port,
```

File: yogstation/code/modules/client/client_procs.dm [hunk 5]
```diff
@@ -5,20 +5,19 @@
...
-	var/sql_computerid = sanitizeSQL(computer_id)
 	var/serverip = "[world.internet_address]"
 
-	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`, `datetime`, `server_ip`, `server_port`, `round_id`, `ckey`, `ip`, `computerid`) VALUES(null, Now(), INET_ATON('[serverip]'), '[world.port]', '[GLOB.round_id]', '[sql_ckey]', INET_ATON('[sql_ip]'), '[sql_computerid]')")
+	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery({"INSERT INTO `[format_table_name("connection_log")]` (`id`, `datetime`, `server_ip`, `server_port`, `round_id`, `ckey`, `ip`, `computerid`)
+	VALUES(null, Now(), INET_ATON(:serverip), :port, :round_id, :ckey, INET_ATON(:address), :computer_id)"},
+	list("serverip" = serverip, "port" = world.port, "round_id" = GLOB.round_id, "ckey" = ckey, "address" = address, "computer_id" = computer_id))
 	if(query_log_connection.Execute(async = TRUE))
 		qdel(query_log_connection)
 		var/datum/DBQuery/query_getid = SSdbcore.NewQuery("SELECT LAST_INSERT_ID();")
```
Comment (alexkar598) [inline yogstation/code/modules/client/client_procs.dm]: This is messing with activity tracking, use query_log_connection.last_insert_id to set the connection number

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/client/client_procs.dm [hunk 6]
```diff
@@ -5,20 +5,19 @@
 	if(!SSdbcore.Connect())
 		return
 
-	var/datum/DBQuery/query_logout = SSdbcore.NewQuery("UPDATE [format_table_name("connection_log")] SET `left` = Now() WHERE id = [number]")
+	var/datum/DBQuery/query_logout = SSdbcore.NewQuery("UPDATE [format_table_name("connection_log")] SET `left` = Now() WHERE id = :number", list("number" = number))
```
Comment (JamieD1) [inline yogstation/code/modules/client/client_procs.dm]:
```suggestion
	var/datum/DBQuery/query_logout = SSdbcore.NewQuery("UPDATE [format_table_name("connection_log")] SET `left` = Now() WHERE id = :number")
```

Comment (JamieD1) [reply yogstation/code/modules/client/client_procs.dm]: Should fix activity

File: .github/workflows/turdis.yml (hunk 2 above)
Comment (Bibby0110) [reply .github/workflows/turdis.yml]: I didn't you just removed PHP while this PR was up :)

File: yogstation/code/modules/client/client_procs.dm (hunk 5 above)
Comment (Bibby0110) [reply yogstation/code/modules/client/client_procs.dm]: Fixed

File: yogstation/code/modules/client/client_procs.dm [hunk 7]
```diff
@@ -5,27 +5,27 @@
...
 	var/serverip = "[world.internet_address]"
 
-	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`, `datetime`, `server_ip`, `server_port`, `round_id`, `ckey`, `ip`, `computerid`) VALUES(null, Now(), INET_ATON('[serverip]'), '[world.port]', '[GLOB.round_id]', '[sql_ckey]', INET_ATON('[sql_ip]'), '[sql_computerid]')")
-	if(query_log_connection.Execute(async = TRUE))
+	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery({"INSERT INTO `[format_table_name("connection_log")]` (`id`, `datetime`, `server_ip`, `server_port`, `round_id`, `ckey`, `ip`, `computerid`)
+	VALUES(null, Now(), INET_ATON(:serverip), :port, :round_id, :ckey, INET_ATON(:address), :computer_id)"},
+	list("serverip" = serverip, "port" = world.port, "round_id" = GLOB.round_id, "ckey" = ckey, "address" = address, "computer_id" = computer_id))
+	if(query_log_connection.Execute())
+		if(query_log_connection.last_insert_id)
+			connection_number = query_log_connection.last_insert_id
```
Comment (alexkar598) [inline yogstation/code/modules/client/client_procs.dm]:
```suggestion
			connection_number = "[num2text(query_log_connection.last_insert_id,24)]"
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]
This is cancer but its the best way to make it not transform the numbers into scientific notation

File: yogstation/code/modules/client/client_procs.dm (hunk 7 above)
Comment (Bibby0110) [reply yogstation/code/modules/client/client_procs.dm]: thanks byond

**Comment by alexkar598** [comment]
This is test merged 24/7 anyways. No point in not merging

**Pull request merged**